### DISCUSSION TITLE
is it necessary for useBottomTabBarHeight to throw outside of the tab navigator?

### DISCUSSION THREAD

**Discussion by abedolinger** [react-navigation/react-navigation#11205] [Q&A]
In our app, we have a navigation flow that goes in and out of our bottom tab bar navigator.  In this flow, we want to use a common hook for some display logic.  However, using the useBottomTabBarHeight hook causes an app crash when used outside the tab bar navigator.

[The code for the useBottomTabBarHeight](https://github.com/react-navigation/react-navigation/blob/main/packages/bottom-tabs/src/utils/useBottomTabBarHeight.tsx) just exposes the context value for the tab bar height if it's available, and throws if it isn't.

So that's my question - why does the hook need to throw outside its intended context?  I think returning `undefined` would be pretty explicit, and would be easy to guard against on our side.

**Comment by ahoopes16** [comment]
I ran into this exact same thing. Would be really nice if it didn't throw an error outside of the context of the bottom tab nav and returned `undefined` or `0` by default.

**Reply by satya164** [reply]
Many people use these hooks incorrectly and open issues even when we have a proper error message. If it returned `undefined` or `0` then it would be much more confusing to anyone who isn't super familiar with it.

If you want to use it in a scenario where it shouldn't throw an error, you still have the ability to import the `BottomTabHeightContext` and use it with `useContext` directly.

**Reply by ahoopes16** [reply]
Fair enough. Thanks for the information!

**Reply by Statsministeriet** [reply]
My solution to this issue:
```typescript
import { BottomTabBarHeightContext } from '@react-navigation/bottom-tabs';

function useBottomTabBarHeight() {
    const height = React.useContext(BottomTabBarHeightContext);

    if (height === undefined) return 0

    return height
}
